source/codegen/templates/service_helpers.mako

```diff
@@ -271,6 +271,205 @@ ${populate_response(function_data=function_data, parameters=parameters)}\
+%>  ::ni::data_monikers::DataMonikerService::RegisterMonikerEndpoint("${moniker_function_name}", ${moniker_function_name});
+</%def>
+
+<%def name="define_moniker_streaming_structs(function_name, function_data, parameters)">\
+<%
+  config = data['config']
+  input_params = [p for p in function_data['parameters'] if common_helpers.is_input_parameter(p)]
+  streaming_param = common_helpers.get_streaming_parameter(function_data['parameters'])
+  service_class_prefix = config["service_class_prefix"]
+  grpc_streaming_type = streaming_param['grpc_streaming_type']
+  struct_name = f"Moniker{function_name.replace('Begin', '')}Data" if function_name.startswith('Begin') else f"Moniker{function_name}Data"
+%>\
+  struct ${struct_name}
+  {
+     % for param in input_params:
+     % if service_helpers.include_param(param, streaming_param):
+     ${param['type']} ${param['name']};
+     % endif
+     % endfor
+     ${service_class_prefix.lower()}_grpc::${grpc_streaming_type} data;
+     std::shared_ptr<${service_class_prefix}LibraryInterface> library;
+  };
+</%def>
+
+<%def name="define_moniker_function_body(function_name, function_data)">\
+<%
+  config = data['config']
+  struct_name = f"Moniker{function_name.replace('Begin', '')}Data" if function_name.startswith('Begin') else f"Moniker{function_name}Data"
+  moniker_function_name = common_helpers.get_data_moniker_function_name(function_name)
+  streaming_param = common_helpers.get_streaming_parameter(function_data['parameters'])
+  streaming_type = streaming_param['type']
+  coerced_type, is_coerced_type_present = service_helpers.get_coerced_type_and_presence(streaming_type)
+  grpc_streaming_type = streaming_param['grpc_streaming_type']
+  is_array = common_helpers.is_array(streaming_type)
+  arg_string = service_helpers.create_args(function_data['parameters'])
+  arg_string = arg_string.replace(", &moniker", "").strip()
+  data_type = streaming_type.replace("[]", "")
+  c_api_name = service_helpers.get_c_api_name(function_name)
+  input_params = [p for p in function_data['parameters'] if common_helpers.is_input_parameter(p)]
+%>\
+::grpc::Status ${moniker_function_name}(void* data, google::protobuf::Arena& arena, google::protobuf::Any& packedData)
+{
+    ${struct_name}* function_data = (${struct_name}*)data;
+    auto library = function_data->library;
+    ${initialize_non_array_parameters(input_params, streaming_param)}\
+    % if streaming_param and streaming_param['direction'] == 'out':
+        ${handle_out_direction(c_api_name, arg_string, data_type, streaming_type, is_coerced_type_present, streaming_param)}\
+    % elif streaming_param and streaming_param['direction'] == 'in':
+        ${handle_in_direction(c_api_name, arg_string, data_type, grpc_streaming_type, streaming_type, coerced_type, is_coerced_type_present, is_array)}\
+    % endif
+    if (status < 0) {
+      std::cout << "${moniker_function_name} error: " << status << std::endl;
```

Comment:
Even in case of error you return `Status::OK` below. Replace this `cout` with `ConvertApiErrorStatusForNiFpga_Session` call similar to how it is being done for other FPGA APIs. Probably calling `populate_error_check` should do that for you.

Reply:
Actually, I think you need to call `populate_response`. The "response" object has `status` field which is supposed to contain `status` returned from driver call. Right now following highlighted status is never returned to caller.

![image](https://github.com/user-attachments/assets/97623f62-ad66-48cc-9bcf-3f91faf8f705)

Reply:
This is a Moniker function, hence there is no response object available in this.
Also, for ConvertApiErrorStatusForNiFpga_Session we need a context object as well but it's not available in the Moniker function.

Reply:
I'll talk to you offline and will raise follow-up based on the discussion.